infrastructure/ai/providers/anthropic_provider.py

```
@@ -22,32 +22,54 @@ def _extract_text_from_content_block(block: Any) -> str:
+
     if isinstance(block, dict):
-        for key in ("text", "content", "value"):
+        for key in ("text", "thinking", "content", "value", "input", "arguments"):
             value = block.get(key)
             if isinstance(value, str) and value.strip():
                 return value
-        if block.get("type") == "json" and block.get("json") is not None:
+            if value is not None and key in {"content", "input", "arguments"} and _is_json_like(value):
+                try:
+                    return json.dumps(value, ensure_ascii=False)
+                except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)